map_model/src/make/buildings.rs

```diff
@@ -369,12 +369,28 @@ fn classify_bldg(
         return BuildingType::Commercial;
     }
 
-    if tags.is("key", "value") {
-        return BuildingType::Residential(42);
+    if tags.is("building", "office") {
+        return BuildingType::Commercial;
+    }
+
+    // TODO - use some "in list" to avoid duplicating
+    if tags.is("building", "garage") {
+        return BuildingType::Empty;
+    }
+    if tags.is("building", "garages") {
+        return BuildingType::Empty;
+    }
+
+    if tags.is("building", "house") {
+        return BuildingType::Residential(3); // I want to use rng.gen_range(1, 5)
```

Comment:
Using a RNG in map building is a bad idea, because importing right now is deterministic. Same code and same input OSM data, same result. This is invaluable for detecting if a code change has an effect or not.

What we can do instead is seed a RNG in a deterministic way, using the OSM IDs. So you can create an RNG like this: `let mut rng = XorShiftRng::seed_from_u64(osm_id);`, where you add a new param to this function `osm_id: i64`, and populate it in `make_all_buildings` from `orig_id.osm_way_id`. You'll also need to add imports at the top: `use rand::{Rng, SeedableRng}; use rand_xorshift::XorShiftRng;`

Reply:
Thanks, I think that I now managed to use it correctly.